test/game/world/chunk_tileTests.cpp

```diff
@@ -4,68 +4,364 @@
 
 #include "game/world/chunk_tile.h"
 
-#include <memory>
-
-#include "proto/resource_entity.h"
-#include "proto/tile.h"
+#include "data/cereal/register_type.h" // Just has to be included somewhere once
+#include "jactorioTests.h"
 
 namespace jactorio::game
 {
-    TEST(ChunkTile, LayerCopy) {
-        // Copying a chunk tile needs to also make a unique copy of unique_data_
-        const auto entity_proto = std::make_unique<proto::ResourceEntity>(proto::ResourceEntity());
+    class ChunkTileTest : public testing::Test
+    {
+    protected:
+        proto::ContainerEntity proto_;
+    };
+
+    TEST_F(ChunkTileTest, CopyTopLeft) {
+        // Since orientation is right, width and height are swapped
+        proto_.SetWidth(2);
+        proto_.SetHeight(5);
+
+        ChunkTile top_left;
+        top_left.SetPrototype(Orientation::right, proto_);
+        top_left.MakeUniqueData<proto::ContainerEntityData>(32);
+
+        const ChunkTile copy{top_left};
+        EXPECT_EQ(copy.GetMultiTileIndex(), 0);
+
+        EXPECT_EQ(copy.GetDimension().span, 5);
+        EXPECT_EQ(copy.GetDimension().height, 2);
+        EXPECT_EQ(copy.GetDimension(), top_left.GetDimension());
+
+        EXPECT_EQ(copy.GetUniqueData<proto::ContainerEntityData>()->inventory.Size(), 32);
+
+        EXPECT_EQ(copy.GetOrientation(), Orientation::right);
+    }
+
+    TEST_F(ChunkTileTest, CopyNonTopLeft) {
+        // Since orientation is right, width and height are swapped
+        proto_.SetWidth(2);
+        proto_.SetHeight(5);
+
+        ChunkTile top_left;
+        ChunkTile tile;
+        tile.SetupMultiTile(4, top_left);
+        tile.SetPrototype(Orientation::right, proto_);
+
+        const ChunkTile copy = tile;
+
+        EXPECT_EQ(copy.GetMultiTileIndex(), 4);
+
+        EXPECT_EQ(copy.GetDimension().span, 5);
+        EXPECT_EQ(copy.GetDimension().height, 2);
+
+        EXPECT_EQ(copy.GetTopLeft(), nullptr); // top left tile not copied
+    }
+
+    TEST_F(ChunkTileTest, CopyAssignTopLeft) {
```

Comment:
No need to test assignment operators for the same operation if the copy and swap idiom is used